### ISSUE TITLE
MetaMask breaks AngularJS websites (including Verizon)

### ISSUE THREAD

**Issue by ptrwtts** [MetaMask/metamask-extension#2500]
Certain websites that use AngularJS, like the My Verizon portal, and younow.com, break when the MetaMask extension is installed.

Steps to reproduce:

- Install MetaMask extension
- Go to MyVerizon: https://myvpostpay.verizonwireless.com/ui/hub/secure/overview
- Login
- The page won't load, and you'll see errors in the console
- If you disable MetaMask and refresh, it works

For younow.com, we found a fix. MetaMask was causing the AngularJS app to lose track of the DOM, so we added a workaround to re-initialize it if MetaMask was detected:

```
if (window.web3 && window.web3.currentProvider.constructor.name == "MetamaskInpageProvider") {
  if ($element[0] == undefined) {
    $element = angular.element(tElement);
  }
}
```

However, this is only a short term fix. I'm sure there are other websites out there that are being impacted. Unfortunately, it's not clear to me what MetaMask does to cause this, but just wanted to bring it to light.

**Comment by bvizzone** [comment]
@ptrwtts 

I am having the same issue. MetaMask is breaking my AngularJS application. Where in the application did you place the above code to work around this issue?

**Comment by pedromcunha** [comment]
Are you using AngularUI Router? @bvizzone 

If so then all you have to do is add the code to the directive variable of the $ViewDirective:

```
var directive = {
    restrict: 'ECA',
    terminal: true,
    priority: 400,
    transclude: 'element',
    compile: function (tElement, tAttrs, $transclude) {
      return function (scope, $element, attrs) {
```

This may change depending on what version of angular ui router you're using, this fix should work for version v0.3.1

**Comment by bvizzone** [comment]
I am using UIRouter 1.0.4 and after a few tests it has not solved the issue. Still debugging/working.

**Comment by Richardxlin** [comment]
+1 I found this issue too. It stops dependency injection of a module in my app.

**Comment by Kaisle** [comment]
+1. Couldn't get the fix to work with UIRouter 1.0.0

**Comment by whymarrh** [comment]
Let us know if anyone is still running into problems with the latest versions of MetaMask and we can reopen this issue.
